wlauto/workloads/googleslides/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,469 @@
+        } catch (UiObjectNotFoundException e) {
+            clickView(BY_ID, "android:id/action_mode_close_button");
+        }
+        SystemClock.sleep(200);
+    }
+
+    public void saveDocument(String docName) throws Exception {
+        timer = new Timer();
+        timer.start();
+        clickView(BY_TEXT, "SAVE");
+        clickView(BY_TEXT, "Device");
+        timer.end();
+        results.put("save_dialog1", timer);
+        // Allow SD card access if requested
+        UiObject permissionView = new UiObject(new UiSelector().textContains("Allow Slides"));
+        if (permissionView.waitForExists(DIALOG_WAIT_TIME_MS)) {
+            clickView(BY_TEXT, "Allow");
+        }
+
+        timer = new Timer();
+        timer.start();
+        UiObject filename = getViewById(PACKAGE_ID + "file_name_edit_text");
+        filename.clearTextField();
+        filename.setText(docName);
+        clickView(BY_TEXT, "Save", CLASS_BUTTON);
+        timer.end();
+        results.put("save_dialog2", timer);
+        // Overwrite if prompted
+        UiObject overwriteView = new UiObject(new UiSelector().textContains("already exists"));
+        if (overwriteView.waitForExists(DIALOG_WAIT_TIME_MS)) {
+            clickView(BY_TEXT, "Overwrite");
+        }
+        sleep(1);
+    }
+
+    public void deleteDocument(String docName) throws Exception {
+        timer = new Timer();
+        timer.start();
+        UiObject doc = getViewByText(docName);
+        doc.longClick();
+        clickView(BY_TEXT, "Remove");
+        timer.end();
+        results.put("delete_dialog1", timer);
+
+        timer = new Timer();
+        timer.start();
+        UiObject deleteButton;
+        try {
+            deleteButton = getUiObjectByText("Remove", CLASS_BUTTON);
+        } catch (UiObjectNotFoundException e) {
+            deleteButton = getUiObjectByText("Ok", CLASS_BUTTON);
+        }
+        deleteButton.clickAndWaitForNewWindow();
+        timer.end();
+        results.put("delete_dialog2", timer);
+        sleep(1);
+    }
+
+    public void repeatClickView(UiObject view, int repeat) throws Exception {
+        if (repeat < 1 || !view.isClickable()) return;
+        while (repeat-- > 0) {
+            view.click();
+            SystemClock.sleep(50); // in order to register as separate click
+        }
+    }
+
+    public UiObject clickView(FindByCriteria criteria, String matching) throws Exception {
+        return clickView(criteria, matching, null, false);
+    }
+
+    public UiObject clickView(FindByCriteria criteria, String matching, boolean wait) throws Exception {
+        return clickView(criteria, matching, null, wait);
+    }
+
+    public UiObject clickView(FindByCriteria criteria, String matching, String clazz) throws Exception {
+        return clickView(criteria, matching, clazz, false);
+    }
+
+    public UiObject clickView(FindByCriteria criteria, String matching, String clazz, boolean wait) throws Exception {
```

Comment:
Is 'View' the best name to use?  These methods are working with UiObjects not Views.  We should consider changing these names